aws/resource_aws_msk_cluster.go

```diff
@@ -587,6 +586,41 @@ func resourceAwsMskClusterUpdate(d *schema.ResourceData, meta interface{}) error
 		}
 	}
 
+	if d.HasChange("kafka_version") {
+		var input *kafka.UpdateClusterKafkaVersionInput
+
+		if d.HasChange("configuration_info") {
+			input = &kafka.UpdateClusterKafkaVersionInput{
+				ClusterArn:         aws.String(d.Id()),
+				ConfigurationInfo:  expandMskClusterConfigurationInfo(d.Get("configuration_info").([]interface{})),
+				CurrentVersion:     aws.String(d.Get("current_version").(string)),
+				TargetKafkaVersion: aws.String(d.Get("kafka_version").(string)),
+			}
+		} else {
+			input = &kafka.UpdateClusterKafkaVersionInput{
+				ClusterArn:         aws.String(d.Id()),
+				CurrentVersion:     aws.String(d.Get("current_version").(string)),
+				TargetKafkaVersion: aws.String(d.Get("kafka_version").(string)),
+			}
+		}
+
+		output, err := conn.UpdateClusterKafkaVersion(input)
+
+		if err != nil {
+			return fmt.Errorf("error updating MSK Cluster (%s) kafka version: %s", d.Id(), err)
```

Comment:
Nit:
```suggestion
			return fmt.Errorf("error updating MSK Cluster (%s) kafka version: %w", d.Id(), err)
```

Reply:
Better to change this in a separate PR as it needs to be changed across all functions for consistency. Happy to modify it here if you prefer.

Reply:
hmm yeah since it's new code, I would prefer we make use of the %w formatter -- future PRs can definitely address the remaining tech-debt 👍

Reply:
done